vector/src/main/java/im/vector/riotx/features/discovery/SettingsEditTextItem.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.riotx.features.discovery
+
+import android.view.inputmethod.EditorInfo
+import android.widget.EditText
+import android.widget.TextView
+import androidx.core.widget.doOnTextChanged
+import com.airbnb.epoxy.EpoxyAttribute
+import com.airbnb.epoxy.EpoxyModelClass
+import com.airbnb.epoxy.EpoxyModelWithHolder
+import com.google.android.material.textfield.TextInputLayout
+import im.vector.riotx.R
+import im.vector.riotx.core.epoxy.VectorEpoxyHolder
+import im.vector.riotx.core.extensions.setTextOrHide
+
+@EpoxyModelClass(layout = R.layout.item_settings_edit_text)
+abstract class SettingsEditTextItem : EpoxyModelWithHolder<SettingsEditTextItem.Holder>() {
+
+    @EpoxyAttribute var descriptionText: String? = null
+    @EpoxyAttribute var errorText: String? = null
+    @EpoxyAttribute var inProgress: Boolean = false
+
+    @EpoxyAttribute
+    var interactionListener: Listener? = null
+
+    override fun bind(holder: Holder) {
+        super.bind(holder)
+        holder.textView.setTextOrHide(descriptionText)
+
+        holder.editText.isEnabled = !inProgress
+
+        if (errorText.isNullOrBlank()) {
+            holder.textInputLayout.error = null
+        } else {
+            holder.textInputLayout.error = errorText
+        }
+
+        holder.editText.doOnTextChanged { code, _, _, _ ->
+            code?.let { interactionListener?.onCodeChange(it.toString()) }
+        }
+        holder.editText.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                interactionListener?.onValidate()
+                return@setOnEditorActionListener true
+            }
+            return@setOnEditorActionListener false
+        }
```

Comment:
we should avoid creating and recreating Listener objects in bind methods. 
Better create it only once

Reply:
Done